e2e/src/components/PostsPreview.ts

```diff
@@ -74,4 +74,51 @@ export default class PostsPreview extends BaseComponent {
         const readTime = await getPreviewReadTime.innerText();
         return readTime;
     }
+
+    get allPosts(): Locator {
+        return this.postLocator;
+    }
+
+    get allReadMoreLinks(): Locator {
+        return this.page.locator('.read-more');
+    }
+
+    public async getAllPosts(): Promise<Locator[]> {
+        return await this.postLocator.all();
+    }
+
+    public async getAllReadMoreLinks(): Promise<Locator[]> {
+        return this.readMoreLinkLocator.all();
+    }
+
+    public async elementsAreVisible(elements: Locator): Promise<void> {
+        for (let i = 0; i < (await elements.count()); i++) {
+            const element = elements.nth(i);
+            await expect(element).toBeVisible();
+        }
+    }
```

Comment:
we have the same logic here >>>
`e2e/src/components/Footer.ts`
<img width="581" alt="Screenshot 2025-02-12 at 09 02 03" src="https://github.com/user-attachments/assets/87cbe3b0-d1d2-495f-8c05-ad0221806576" />